chromium_src/chrome/browser/ui/views/tabs/tab_hover_card_bubble_view.cc

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "base/strings/string_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "chrome/browser/ui/views/tabs/tab.h"
+#include "chrome/browser/ui/views/tabs/tab_hover_card_bubble_view.h"
+#include "content/public/common/url_constants.h"
+#include "ui/views/controls/label.h"
+
+#define TabHoverCardBubbleView TabHoverCardBubbleView_ChromiumImpl
+#include "../../../../../../chrome/browser/ui/views/tabs/tab_hover_card_bubble_view.cc"
+#undef TabHoverCardBubbleView
+
+class TabHoverCardBubbleView_ChromiumImpl::BraveAnonymous {
+ public:
+  static void UpdateCardContent(TabHoverCardBubbleView_ChromiumImpl* base,
+                                const Tab* tab) {
+    base->TabHoverCardBubbleView_ChromiumImpl::UpdateCardContent(tab);
+    const base::string16& domain = base->domain_label_->GetText();
+    const base::string16 kChromeUISchemeU16 =
+        base::ASCIIToUTF16(content::kChromeUIScheme);
+    // Replace chrome:// with brave://. Since this is purely in the UI we can
+    // just do a sub-string replacement instead of parsing into GURL.
+    if (base::StartsWith(domain, kChromeUISchemeU16,
+                         base::CompareCase::INSENSITIVE_ASCII)) {
+      base::string16 new_domain = domain;
+      base::ReplaceFirstSubstringAfterOffset(
+          &new_domain, 0ul, kChromeUISchemeU16,
+          base::ASCIIToUTF16(content::kBraveUIScheme));
+      base->domain_label_->SetText(new_domain);
+    }
+  }
+};
+
+void TabHoverCardBubbleView::UpdateCardContent(const Tab* tab) {
+  TabHoverCardBubbleView_ChromiumImpl::BraveAnonymous::UpdateCardContent(
```

Comment:
re https://github.com/brave/brave-core/pull/3687/files#r335575232 - why aren't we just directly calling the code in `TabHoverCardBubbleView_ChromiumImpl::BraveAnonymous::UpdateCardContent`?

Reply:
Similarly here, I am going through the inner class method to get to the private members of the base class (because I couldn't make the subclass a friend with this approach).

Reply:
why rename it? Wouldn't it be simpler to subclass/friend and override in tab_strip.cc/.h? You need a patch either way and it would be simpler to understand imo

Reply:
Mainly was just trying out a way to make changes without having to patch tab_strip.cc.

Reply:
you wouldn't have to patch tab_strip.cc, you can change the class name with a chromium_src override

Reply:
we already do that with `NewTabButton` -> `BraveNewTabButton` in an existing override for tab_strip.cc

Reply:
Yes, but with `TabHoverCardBubbleView`, tab_strip.cc:126 also has
`  TabHoverCardEventSniffer(TabHoverCardBubbleView* hover_card,
                           TabStrip* tab_strip)`
which would get redefined.

Reply:
why does that matter? I saw it, but maybe I'm missing something?

Reply:
ok, I see why it matters (`TabHoverCardBubbleView* hover_card_ = nullptr;`) in the header, but can't you just override in the header as well?

Reply:
I mean, what you have works, it's just harder to follow because it's very different from what we would normally do in this case. If we can subclass and override I think it would be better for long-term maintenance

Reply:
or if we went this route, wouldn't it be simpler to just directly patch in a method to `TabHoverCardBubbleView_ChromiumImpl` instead of creating an inner class?